pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -450,33 +450,50 @@ func managePod(ctx context.Context, configMapsGetter corev1client.ConfigMapsGett
 		}
 	}
 
-	containerArgsWithLoglevel := required.Spec.Containers[0].Args
-	if argsCount := len(containerArgsWithLoglevel); argsCount > 1 {
-		return nil, false, fmt.Errorf("expected only one container argument, got %d", argsCount)
-	}
-	if !strings.Contains(containerArgsWithLoglevel[0], "exec hyperkube kube-controller-manager") {
-		return nil, false, fmt.Errorf("exec hyperkube kube-controller-manager not found in first argument %q", containerArgsWithLoglevel[0])
-	}
-
-	containerArgsWithLoglevel[0] = strings.TrimSpace(containerArgsWithLoglevel[0])
+	// This section sets the log levels for all containers that take a "1-line" argument
+	logLevel := 2
 	switch operatorSpec.LogLevel {
 	case operatorv1.Normal:
-		containerArgsWithLoglevel[0] += fmt.Sprintf(" -v=%d", 2)
+		logLevel = 2
 	case operatorv1.Debug:
-		containerArgsWithLoglevel[0] += fmt.Sprintf(" -v=%d", 4)
+		logLevel = 4
 	case operatorv1.Trace:
-		containerArgsWithLoglevel[0] += fmt.Sprintf(" -v=%d", 6)
+		logLevel = 6
 	case operatorv1.TraceAll:
-		containerArgsWithLoglevel[0] += fmt.Sprintf(" -v=%d", 8)
+		logLevel = 8
 	default:
-		containerArgsWithLoglevel[0] += fmt.Sprintf(" -v=%d", 2)
+		logLevel = 2
+	}
+	// containers[0] = kube-controller-manager
+	// containers[1] = cluster-policy-controller
+	// containers[2] = kube-controller-manager-cert-syncer
+	// containers[3] = kube-controller-manager-recovery-controller
+	for i := 0; i < len(required.Spec.Containers); i++ {
+		if i == 2 {
```

Comment:
Can you add here also name check of the container, just to be extra careful, this is very delicate :cold_sweat: